### ISSUE TITLE
Optional parentheses

### ISSUE THREAD

**Issue by cedvdb** [dart-lang/language#4010]
Rust has them. I did not see an issue regarding optional parentheses. 

```dart
    if money < 0 {
        print!('You are poor');
        throw UselessUserException();
    } else if money  > 0 {
        print!('Damn');
    } else {
        print!('Doing all right I see');
    }
```

Just a tiny bit easier to read, but rust requires brackets.

**Comment by ykmnkmi** [comment]
What about this?
```dart
if () == () {
  // ...
}
```

**Comment by mateusfccp** [comment]
I find it actually harder to read.

It's subjective, after all.

**Comment by mateusfccp** [comment]
> What about this?
> 
> ```dart
> if () == () {
>   // ...
> }
> ```

It's not impossible to disambiguate this expression, but I don't think it's worth the hassle.

**Comment by cedvdb** [comment]
> It's subjective, after all.

Well there is fewer characters once there is more than 1 line inside the condition, but I get you point. I wanted to gauge if people felt like me. Not the case, judging from the downvotes. I'm going to close this then

**Comment by mateusfccp** [comment]
> Well there is fewer characters once there is more than 1 line inside the condition, but I get you point. I wanted to gauge if people felt like me. The downvotes have spoken. I'm going to close this then

Yes, as a general rules, fewer characters carry less clutter and is usually more readable, but I don't think that's always the case.

In my case, I find it easier to quickly identify the expressions because they are always between `(` and `)`. By not having them, it's like the expression mixes up with the keyword. But as I said, it's something subjective, and I'm sure it's more related to what one is used to.

> Not sure if you were talking about optional parentheses or not, but it would be optional so no breaking change.

I considered both cases in my response.
1. If we make them optional, we will have more problems with ambiguity.
2. If we make them mandatory, it will be a huge breaking change.
